dali/operators/reader/video_reader_op.h

```diff
@@ -26,11 +26,16 @@ namespace detail {
 inline int VideoReaderOutputFn(const OpSpec &spec) {
   std::string file_root = spec.GetArgument<std::string>("file_root");
   std::string file_list = spec.GetArgument<std::string>("file_list");
+  std::vector<std::string> file_names = spec.GetRepeatedArgument<std::string>("filenames");
+  std::vector<int> labels;
+  bool has_labels_arg = spec.TryGetRepeatedArgument(labels, "labels");
   bool enable_frame_num = spec.GetArgument<bool>("enable_frame_num");
   bool enable_timestamps = spec.GetArgument<bool>("enable_timestamps");
   int num_outputs = 1;
-  if (!file_root.empty() || !file_list.empty()) {
-    num_outputs++;
+  if ((!file_names.empty() && has_labels_arg) || file_names.empty()) {
```

Comment:
I don't quite understand this condition.
If either you have file names and labels have been requested, or if filenames is empty. Why would you want labels unconditionally when file names is empty?

Reply:
Why not just 
```if (has_labels_arg)```
If you requested labels output, you will get one, even if its empty.

Reply:
It applies:
- when you have a file_list provided
- you have file_root provided
- you have filenames and has_labels_arg

I can replace `file_names.empty()` with `!file_root.emtpy() || !file_list.empt()`